rasa/core/brokers/celery.py

```diff
@@ -0,0 +1,56 @@
+import logging
+from functools import partial
+from asyncio import AbstractEventLoop
+from typing import Dict, Text, Callable, Any, Optional, Union
+
+from celery import Celery
+
+import rasa.shared.utils.common
+from rasa.core.brokers.broker import EventBroker
+from rasa.utils.endpoints import EndpointConfig
+
+logger = logging.getLogger(__name__)
+
+
+class CeleryEventBroker(EventBroker):
+    def __init__(
+        self,
+        broker_url: Optional[Text] = None,
+        task_name: Text = None,
+        **kwargs: Dict[Text, Any],
+    ) -> None:
+        """Initializes `CeleryBrokerEvent`."""
+        self._task = self._get_task(broker_url, task_name)
+        self._task_kwargs: Dict[Text, Any] = kwargs
+
+    def _get_task(
+        self, broker_url: Optional[Text], task_name: Text
+    ) -> Callable[..., None]:
+        """Return the task based on the provided broker_url and task_name."""
+        if broker_url:
+            task_instance: Celery = Celery("rasa_event_broker", broker=broker_url)
+            task_callable = partial(task_instance.send_task, task_name)
+        else:
+            task_instance = rasa.shared.utils.common.class_from_module_path(task_name)
```

Comment:
This seems redundant as Celery requires URL to be present.
If `broker_url` is not present we need to issue RasaException to signal to the user that configuration they provided is not valid.

Reply:
@radovanZRasa, the idea here would be to allow users to have the choice to either use `send_task` or `apply_async`. If the user just uses apply_async, they would not need to pass it the broker_url. I did find that passing in the broker_url could be a bit of a hassle from the user's side as the user would need to pass it into the endpoint.yaml file via env variables. This is because the broker_url should contain sensitive auth info

If the user uses apply_async, they can use the broker_url that is already set functionally nearby their function. What do you think?